packages/editor/src/lib/components/Shape.tsx

```diff
@@ -124,7 +124,8 @@ export const Shape = memo(function Shape({
 			const shape = editor.getShape(id)
 			if (!shape) return // probably the shape was just deleted
 
-			const isCulled = editor.isShapeCulled(shape)
+			const culledShapes = editor.getCulledShapes()
+			const isCulled = culledShapes.has(id)
 			setStyleProperty(containerRef.current, 'display', isCulled ? 'none' : 'block')
 			setStyleProperty(bgContainerRef.current, 'display', isCulled ? 'none' : 'block')
```

Comment:
i did some perf testing. We run these `setStyleProperty` on every frame for every shape while panning, even when the isCulled value hasn't changed, and it seems to take up quite a lot of time. You could fix that by using a useRef to store the previous isCulled value and returning early if it didn't change.

Reply:
I can add this, I did this once before

Reply:
Done!